src/os/bluestore/BlueStore.cc

```diff
@@ -3506,7 +3506,14 @@ BlueStore::BlobRef BlueStore::ExtentMap::split_blob(
 void BlueStore::Onode::get() {
   if (++nref >= 2 && !pinned) {
     OnodeCacheShard* ocs = c->get_onode_cache();
-    std::lock_guard l(ocs->lock);
+    ocs->lock.lock();
+    OnodeCacheShard* ocs1 = c->get_onode_cache();
+    if (ocs1 != ocs) {
+      ceph_assert("ocs1 != ocs : split_cache problem");
```

Comment:
This to be removed for production I presume?

Reply:
@ifed01 True. It is not for production.
Thanks for DNM. This code is not do-while because it is intended for bug reconnaissance.